File: packages/ai-config/src/schema/provider.schema.ts
// Esquema de validación para Provider
import { z } from 'zod';

export const SecretRefSchema = z.union([
  z.string(),
  z.object({
    type: z.enum(['env', 'file', 'exec']),
    name: z.string().optional(),
    path: z.string().optional(),
    command: z.string().optional(),
  }),
]);

export const AuthSourceSchema = z.object({
  mode: z.enum(['api_key', 'oauth', 'secret_ref']),
  value: z.string().optional(),
  token: z.string().optional(),
  ref: SecretRefSchema.optional(),
});

export const ProviderSchema = z.object({
  id: z.string(),
  name: z.string(),
  type: z.string(),
  models: z.array(z.string()).optional(),
  auth: AuthSourceSchema,
  authEnvVar: z.string().optional(),
  profileKey: z.string().optional(),
  priority: z.number().optional(),
  meta: z.record(z.string(), z.unknown()).optional(),
});

File: packages/ai_config/schema/provider_schema.py
from typing import Any, Dict, List, Literal, Optional, Union

from pydantic import BaseModel


class SecretRefSchema(BaseModel):
    type: Literal['env', 'file', 'exec']
    name: Optional[str] = None
    path: Optional[str] = None
    command: Optional[str] = None


class AuthSourceSchema(BaseModel):
    mode: Literal['api_key', 'oauth', 'secret_ref']
    value: Optional[str] = None
    token: Optional[str] = None
    ref: Optional[Union[str, SecretRefSchema]] = None


class ProviderSchema(BaseModel):
    id: str
    name: str
    type: str
    models: Optional[List[str]] = None
    auth: AuthSourceSchema
    auth_env_var: Optional[str] = None
    profile_key: Optional[str] = None
    priority: Optional[int] = None
    meta: Optional[Dict[str, Any]] = None
